src/gen9_avc_encoder.c

```diff
@@ -3596,6 +3662,456 @@ gen9_avc_set_curbe_mbenc(VADriverContextP ctx,
+            break;
+        case SLICE_TYPE_B:
+            memcpy(cmd, gen9_avc_fei_mbenc_curbe_b_frame_init_data, sizeof(gen9_avc_fei_mbenc_curbe_data));
+            break;
+        default:
+            assert(0);
+        }
+
+    }
+    /* 4 means full search, 6 means diamand search */
+    me_method  = (fei_param->search_window == 5) ||
+                 (fei_param->search_window == 8) ? 4 : 6;
+
+    ref_width    = fei_param->ref_width;
+    ref_height   = fei_param->ref_height;
+    len_sp       = fei_param->len_sp;
+    /* If there is a serch_window, discard user provided ref_width, ref_height
+     * and search_path length */
+    switch (fei_param->search_window) {
+    case 0:
+        /*  not use predefined search window, there should be a search_path input */
+        if ((fei_param->search_path != 0) &&
+            (fei_param->search_path != 1) &&
+            (fei_param->search_path != 2)) {
+            WARN_ONCE("Invalid input search_path for SearchWindow=0  \n");
+            assert(0);
+        }
+        /* 4 means full search, 6 means diamand search */
+        me_method = (fei_param->search_path == 1) ? 6 : 4;
+        if (((ref_width * ref_height) > 2048) || (ref_width > 64) || (ref_height > 64)) {
+            WARN_ONCE("Invalid input ref_width/ref_height in"
+                      "SearchWindow=0 case! \n");
+            assert(0);
+        }
+        break;
+
+    case 1:
+        /* Tiny - 4 SUs 24x24 window */
+        ref_width  = 24;
+        ref_height = 24;
+        len_sp     = 4;
+        break;
+
+    case 2:
+        /* Small - 9 SUs 28x28 window */
+        ref_width  = 28;
+        ref_height = 28;
+        len_sp     = 9;
+        break;
+    case 3:
+        /* Diamond - 16 SUs 48x40 window */
+        ref_width  = 48;
+        ref_height = 40;
+        len_sp     = 16;
+        break;
+    case 4:
+        /* Large Diamond - 32 SUs 48x40 window */
+        ref_width  = 48;
+        ref_height = 40;
+        len_sp     = 32;
+        break;
+    case 5:
+        /* Exhaustive - 48 SUs 48x40 window */
+        ref_width  = 48;
+        ref_height = 40;
+        len_sp     = 48;
+        break;
+    case 6:
+        /* Diamond - 16 SUs 64x32 window */
+        ref_width  = 64;
+        ref_height = 32;
+        len_sp     = 16;
+        break;
+    case 7:
+        /* Large Diamond - 32 SUs 64x32 window */
+        ref_width  = 64;
+        ref_height = 32;
+        len_sp     = 32;
+        break;
+    case 8:
+        /* Exhaustive - 48 SUs 64x32 window */
+        ref_width  = 64;
+        ref_height = 32;
+        len_sp     = 48;
+        break;
+
+    default:
+        assert(0);
+    }
+
+    /* ref_width*ref_height = Max 64x32 one direction, Max 32x32 two directions */
+    if (is_bframe) {
+        if (ref_width > 32) {
+            ref_width  = 32;
+        }
+        if (ref_height > 32) {
+            ref_height = 32;
+        }
+    }
+
```

Comment:
As the comments above: "/* ref_width*ref_height = Max 64x32 one direction, Max 32x32 two directions */". Seems the height needs to be clipped to 32, I guess, : )